lib/cagen.py

```diff
@@ -125,30 +129,32 @@ def usercert(self, username, password, days=None):
         globus_dir = os.path.join(os.path.expanduser('~' + username), '.globus')
         user_path = os.path.join(globus_dir, 'usercert.pem')
         user_keypath = os.path.join(globus_dir, 'userkey.pem')
+        new_keypath = user_keypath + '.new'
+        user = pwd.getpwnam(username)
         user_subject = self._subject_base + '/OU=People/CN=' + username
         user_request = 'user_req'
 
         try:
             os.makedirs(globus_dir, 0755)
```

Comment:
For future-proofing, consider changing octal constants from `0XXX` to `0oXXX`. (The former syntax is not supported in py3.)